torch/csrc/jit/mobile/interpreter.cpp

```diff
@@ -43,8 +45,17 @@ bool InterpreterState::run(Stack& stack) {
             mobile_debug_info->setOpIdx(pc);
           }
         }
-        RECORD_FUNCTION(code_->op_names_[inst.X].name, stack);
 #endif
+        //TODO(iliacher): remove the workaround after RecordFunction is in Dispatcher
+        bool prev_value = observersEnabled();
```

Comment:
nit: this could be simpler by writing

```cpp
bool prev_value = observersEnabled();
enableObservers(true);
RECORD_FUNCTION(code_->op_names_[inst.X].name, stack);
enableObservers(prev_value);
```

but the current way is fine too, up to you.
Also, I'm not sure how this value is stored. If `enableObservers` modifies a non-thread-local or an atomic variable, then your approach is better because it avoids modifying that.

Reply:
I wanted to avoid the unnecessary calls, enableObservers sets dispatcher TLS key (at the moment, when we move RecordFunction in dispatcher it will set a separate TLS flag)